run.py

```diff
@@ -122,36 +125,32 @@ def config_validator():
-        # To avoid this, we open the file in try catch - just to be sure
         if os.path.isfile(os.path.join(os.getcwd(), 'issues.db')):
-            try:
-                with open(os.path.join(os.getcwd(), 'issues.db')) as file:
-                    if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
-                        print 'backup db folder does not exist'
-                        os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
-                    else:
-                        print 'backup db folder exists'
-                        # Retain last 3 recent backup files
-                        no_of_backup_files = len(os.listdir(app.config['BACKUP_DEFAULT_DEST']))
-                        if no_of_backup_files > 3:
-                            for idx, val in enumerate(os.listdir(app.config['BACKUP_DEFAULT_DEST'])):
-                                if idx < no_of_backup_files - 2:
-                                    os.remove(app.config['BACKUP_DEFAULT_DEST'] + val)
-                    current_ts = time.time()
-                    time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
-                    issue_backup_db = 'issues_backup_' + time_stamp + '.db'
-                    os.rename(os.getcwd() + '/issues.db', app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
-                    try:
-                        os.remove(os.path.join(os.getcwd(), 'issues.db'))
-                    except OSError:
-                        print 'Backup issues.db complete'
-
-            except IOError:
-                sys.exit(FILE_NOT_FOUND_ERROR)
-
+            issue_engine = create_engine('sqlite:///' + os.path.join(
+                app.config['BASE_DIR'], 'issues.db'))
+            issue_session_maker = sessionmaker(autocommit=False,
+                                               autoflush=False,
+                                               bind=issue_engine)
+            issue_db = scoped_session(issue_session_maker)
+            # Take a backup if issues.db has data dump.
+            if issue_db().execute('select count(*) from webcompat_issues').fetchall()[0][0] > 0:
+                if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
+                    print 'Backup db folder does not exist'
+                    os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
+                else:
+                    print 'Backup db folder exists'
+                current_ts = time.time()
+                time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
+                issue_backup_db = 'issues_backup_' + time_stamp + '.db'
+                os.rename(os.getcwd() + '/issues.db', app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
+                # Retain last 3 recent backup files
+                no_of_backup_files = len(os.listdir(app.config['BACKUP_DEFAULT_DEST']))
```

Comment:
from the documentation of `os.listdir`

> Return a list containing the names of the entries in the directory given by path. **The list is in arbitrary order.** It does not include the special entries '.' and '..' even if they are present in the directory.

There's no guarantee that the list you are building will be in time order or alphabetical order. It's very likely but not mandatory. It's also not necessary to get the length of the list. It's often a sign of a JavaScriptism :) List are enumerable properties.